docs/dcs_failsafe_mode.rst

```diff
@@ -52,6 +53,10 @@ F.A.Q.
 - What if all members of the Patroni cluster are lost while DCS is down?
 
   Patroni could be configured to create the new replica from the backup even when the cluster doesn't have a leader. But, if the new member isn't present in the ``/failsafe`` key, it will not be able to grab the leader lock and promote.
+  
+- What will happen if the primary lost access to DCS while replicas didn't?
+
+  The primary will execute the failsafe code and contact all known replicas. These replicas will use this information as an indicator that the primary is alive and will not start the leader race even if the leader lock in DCS has expired.
```

Comment:
I guess, I have already asked it but once again. Could it be that the leader lock expires earlier, than the leader contacts all known replicas?

Reply:
In theory - yes (especially if someone screwed up the configuration).

But, it isn't dangerous. When the replica does the leader race it also contacts the former primary via REST API. Once it noticed that the "old" primary is still alive it will not try to grab the leader lock.